src/main/java/io/github/cottonmc/resources/CommonResources.java

```diff
@@ -0,0 +1,97 @@
+        builtinMetal("lead",   BlockSuppliers.IRON_TIER_ORE,  MACHINE_AFFIXES);
+        builtinMetal("zinc",   BlockSuppliers.STONE_TIER_ORE);
+        builtinMetal("uranium",BlockSuppliers.IRON_TIER_ORE); //TODO: Registers ingot, nugget, and block. Should it?
+        
+        builtinMetal("steel",    null, MACHINE_AFFIXES);
+        builtinMetal("brass",    null, MACHINE_AFFIXES);
+        builtinMetal("electrum", null); //no ore, no gears/plates
+        
+        builtinItem("coal_dust");
+        builtinItem("coal_coke"); //TODO: Define a non-metal type we can add a blind block supplier to?
+        builtinItem("mercury");
+        
+        //These might get rods or molten capsules. They'd just need to be added to the end.
+        builtinItem("uranium",   RADIOACTIVE_AFFIXES);
+        builtinItem("plutonium", RADIOACTIVE_AFFIXES);
+        builtinItem("thorium",   RADIOACTIVE_AFFIXES);
+    }
+
+    private static void builtinMetal(String id, Supplier<Block> oreSupplier, String... extraAffixes) {
+        MetalResourceType result = new MetalResourceType(id).withOreSupplier(oreSupplier);
+        if (extraAffixes.length>0) result.withItemAffixes(extraAffixes);
+        
+        BUILTINS.put(id, result);
+    }
+
+    private static void builtinItem(String id, String... extraAffixes) {
+        ItemResourceType result = new ItemResourceType(id, extraAffixes);
+        if (extraAffixes.length==0) result.withAffixes(""); //This is just a base type with no affixes, like "mercury".
+        BUILTINS.put(id, result);
+    }
+
+    /**Registers an Item based on its name if it is supported or returns it if it already exists.
+    * @param name The name of the block that shall be registered.
+     * @return The Block associated with the given name or null if no Block for that name was found
+     */
+    @Nullable
+    public static Block provideBlock(String name){
+        Identifier id = new Identifier("cotton", name);
+        if (Registry.BLOCK.contains(id)) {
+            //It exists, get it from the registry
+            return Registry.BLOCK.get(id);
+        }
+        
+        for(ResourceType type : BUILTINS.values()) {
+            if (type.governs(name)) return type.getBlock(name);
+        }
+        
+        return null;
+    }
+
+    /**Registers an Item based on its name if it is supported, or returns it if it already exists.
+     * @param name The name of the item
+     * @return The Item associated with the given name or null if the name is not supported.
+     */
+    @Nullable
+    public static Item provideItem(String name){
+        Identifier id = new Identifier("cotton", name);
+        if (Registry.ITEM.contains(id)) {
+            //It exists, get it from the registry
+            return Registry.ITEM.get(id);
+        }
+        
+        for(ResourceType type : BUILTINS.values()) {
+            if (type.governs(name)) return type.getItem(name);
+        }
+        
+        return null;
+    }
+
```

Comment:
Maybe I am missing something, but I still don't see a function that adds all items for a domain. e.g. for "copper" it should add copper_ingot, copper_ore,...

Reply:
For that you'd do
```java
CommonResources.provideResource("copper").registerAll();
```

Reply:
Totally missed that there was a provideResource method. Yeah, that makes sense.